mesonbuild/dependencies/ui.py

```diff
@@ -254,13 +251,10 @@ def gen_bins():
             else:
                 arg = ['-v']
 
-            # Ensure that the version of qt and each tool are the same
-            _, out, err = mesonlib.Popen_safe(p.get_command() + arg)
-            if b.startswith('lrelease') or not self.version.startswith('4'):
-                care = out
-            else:
-                care = err
-            if mesonlib.version_compare(self.version, '== {}'.format(care.split(' ')[-1])):
+            p = interp_obj.find_program_impl([b], silent=True, required=required,
+                                             version_arg=arg,
+                                             wanted=wanted).held_object
```

Comment:
And actually, if we do that, we don't even need the `found` dictionary at all, do we?

Reply:
That was my first idea too, but not without refactoring find_program_impl()... because currently it only checks the version against the first name it found. I guess that's really corner case, but you could have libqt 5.x, moc-qt5 5.y and moc 5.x, it won't check for moc when version mismatch for moc-qt5... dunno if we care...